lib/dry/schema/path.rb

```diff
@@ -60,84 +60,33 @@ def initialize(keys)
 
       # @api private
       def to_h(value = EMPTY_ARRAY.dup)
-        curr_idx = 0
-        last_idx = keys.size - 1
-        hash = EMPTY_HASH.dup
-        node = hash
-
-        while curr_idx <= last_idx
-          node =
-            node[keys[curr_idx]] =
-              if curr_idx == last_idx
-                value.is_a?(Array) ? value : [value]
-              else
-                EMPTY_HASH.dup
-              end
-
-          curr_idx += 1
-        end
+        value = [value] unless value.is_a?(Array)
 
-        hash
+        keys.reverse_each.reduce(value) { |result, key| {key => result} }
       end
 
       # @api private
       def each(&block)
         keys.each(&block)
       end
 
-      # @api private
-      def index(key)
-        keys.index(key)
-      end
-
-      def without_index
```

Comment:
It's not marked as `@private`, but it's unused in dry-s/dry-v and I'm not sure if it's intended to be public.